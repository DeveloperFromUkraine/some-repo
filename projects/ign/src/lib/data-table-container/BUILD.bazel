load("@angular//:index.bzl", "ng_module")
load("@io_bazel_rules_sass//:defs.bzl", "sass_binary")

package(default_visibility = ["//visibility:public"])

sass_binary(
    name = "style",
    src = "data-table-container.scss",
)

ng_module(
    name = "data-table-container-module",
    srcs = [
        "data-table-container.module.ts",
        "data-table-container.component.ts",
    ],
    assets = [
        ":data-table-container.css",
        "data-table-container.component.html"
        ],
    tsconfig = "//:tsconfig.json",
    deps = [
        "@npm//tslib",
        "@npm//@types",
        "@angular//packages/core",
        "@angular//packages/common",
        "@angular_material//src/lib:material"
    ]
)
