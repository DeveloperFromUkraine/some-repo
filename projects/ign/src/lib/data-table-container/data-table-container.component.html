<ng-content></ng-content>

<mat-paginator #paginator
               [pageSize]="(pageSize || 10)"
               [pageSizeOptions]="(pageSizeOptions || [5, 10, 15, 20, 50])"
               [pageIndex]="pageIndex"
               [length]="totalElements"
               (page)="emitPageEvents($event)">
</mat-paginator>
