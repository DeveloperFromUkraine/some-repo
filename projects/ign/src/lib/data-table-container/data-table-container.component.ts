import { Component, Input, Output, ViewChild, AfterViewInit } from '@angular/core';
import { MatPaginator, MatTableDataSource } from '@angular/material';
import { BehaviorSubject } from 'rxjs';

@Component({
  selector: 'ign-data-table-container',
  templateUrl: 'data-table-container.component.html',
  styleUrls: ['./data-table-container.css'],
})
export class DataTableContainerComponent implements AfterViewInit {
  @Input() dataSource: MatTableDataSource<any>;
  @Input() pageIndex: number;
  @Input() totalElements: number;
  @Input() pageSizeOptions: number[];
  @Input() pageSize: number;

  @Output() paginatorEvents$ = new BehaviorSubject<any>({});

  @ViewChild(MatPaginator) paginator: MatPaginator;

  ngAfterViewInit() {
    // Set the paginator after the view init since this component will
    // be able to query its view for the initialized paginator.
    this.dataSource.paginator = this.paginator;
  }

  emitPageEvents(event: any) {
    this.paginatorEvents$.next(event);
  }
}
