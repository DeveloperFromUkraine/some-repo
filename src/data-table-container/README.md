# Data Table Container

A container to house a Material data table. This container provides a paginator at the container's footer.

## API
  ### Selector
    * ign-data-table-container

  ### Export As
    * DataTableContainerComponent

  ### Properties 
    * `@Input() dataSource: MatTableDataSource<any>`
      * Data being used for Material data table 

    * `@Input() pageIndex: number`
      * The index of the page

    * `@Input() totalElements: number`
      * The total elements in a data table 

    * `@Input() pageSizeOptions: number[]`
      * Page size options to be made available within the paginator. If none are provided, [5, 10, 15, 20, 50] will be used

    * `@Input() pageSize: number`
      * Default number of elements to display on a page. If none are specified, 10 will be displayed
      
    * `@Output() paginatorEvents$: BehaviorSubject<any>({});`
      * Emits corresponding events from paginator interaction


## Implementation HTML
  <div class="example-container mat-elevation-z8">
    <ign-data-table-container [dataSource]="dataSource">
        <mat-table #table [dataSource]="dataSource">
          <!--- Note that these columns can be defined in any order.
                The actual rendered columns are set as a property on the row definition" -->
            <!-- Name Column -->
            <ng-container matColumnDef="name">
            <mat-header-cell \*matHeaderCellDef> Name </mat-header-cell>
            <mat-cell \*matCellDef="let element"> {{element.name}} </mat-cell>
            </ng-container>             
          <!-- Position Column -->
          <ng-container matColumnDef="position">
            <mat-header-cell \*matHeaderCellDef> Position </mat-header-cell>
            <mat-cell \*matCellDef="let element"> {{element.position}} </mat-cell>
          </ng-container>              
          <!-- Type Column -->
          <ng-container matColumnDef="type">
            <mat-header-cell \*matHeaderCellDef> Type </mat-header-cell>
            <mat-cell \*matCellDef="let element"> {{element.type}} </mat-cell>
          </ng-container>
          <mat-header-row \*matHeaderRowDef="displayedColumns"></mat-header-row>
          <mat-row \*matRowDef="let row; columns: displayedColumns;"></mat-row>
        </mat-table>
      </ign-data-table-container>
  </div>

## Implementation TS
  displayedColumns = ['name', 'position', 'type'];
  dataSource = new MatTableDataSource<Element>(ELEMENT_DATA);

  export interface Element {
    name: string;
    position: string;
    type: string;
  }

let ELEMENT_DATA: Element[] = [
    {name: 'Maham Boghani', position: 'Software Engineer', type: 'Human'},
    {name: 'Watson', position: 'Artificial Intelligence', type: 'Computer'},
    {name: 'Andy Avalos', position: 'Electrical Engineer', type: 'Human'},
    {name: 'Jensen', position: 'Director of Sleep', type: 'House Cat'},
    {name: 'Mariah Avalos', position: 'Software Engineer', type: 'Introvert'},
    {name: 'Emma Bear', position: 'Destroyer Of String', type: 'House Cat'},
    {name: 'Uchenna Ohaeto', position: 'Software Engineer', type: 'Human'},
    {name: 'Dennis', position: 'Being Repressed', type: 'Human'},
    {name: 'Sophia', position: 'Home Security', type: 'Dog'},
    {name: 'Brian Mahecha', position: 'Software Test Engineer', type: 'Human'},
    {name: 'Spock', position: 'First Officer, USS Enterprise', type: 'Vulcan'},
    {name: 'Daniel Weisberg', position: 'Software Engineer', type: 'Human'},
    {name: 'Inigo Montoya', position: 'Seeking Revenge', type: 'Human'},
    {name: 'Miguel Conde', position: 'Software Engineer', type: 'Human'},
    {name: 'Nikola Tesla', position: 'Inventor', type: 'Human'},
  
];