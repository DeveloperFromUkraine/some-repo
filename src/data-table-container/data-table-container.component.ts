import { Component, Input, Output, ViewChild, AfterViewInit } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { MatPaginator } from '@angular/material/paginator';
import { MatTableDataSource } from '@angular/material/table';

@Component({
  selector: 'ign-data-table-container',
  templateUrl: 'data-table-container.component.html',
  styleUrls: ['./data-table-container.scss'],
})
export class DataTableContainerComponent implements AfterViewInit {
  @Input() dataSource: MatTableDataSource<any>;
  @Input() pageIndex: number;
  @Input() totalElements: number;
  @Input() pageSizeOptions: number[];
  @Input() pageSize: number;

  @Output() paginatorEvents$ = new BehaviorSubject<any>({});

  @ViewChild(MatPaginator) paginator: MatPaginator;

  ngAfterViewInit() {
    // Set the paginator after the view init since this component will
    // be able to query its view for the initialized paginator.
    this.dataSource.paginator = this.paginator;
  }

  emitPageEvents(event: any) {
    this.paginatorEvents$.next(event);
  }
}
