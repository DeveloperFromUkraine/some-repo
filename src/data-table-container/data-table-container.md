*** HTML ***

# Data Table Container

## Use
A container to house a Material data table. This container provides a paginator at the container's footer.

## Sample
<mat-tab-group>
    <mat-tab label="Component Sample">
        <div class="tab-height">
                <div class="example-container mat-elevation-z8">
                    <ign-data-table-container [dataSource]="dataSource">
                        <mat-table #table [dataSource]="dataSource">
                          <!--- Note that these columns can be defined in any order.
                                The actual rendered columns are set as a property on the row definition" -->
                            <!-- Name Column -->
                           <ng-container matColumnDef="name">
                            <mat-header-cell \*matHeaderCellDef> Name </mat-header-cell>
                            <mat-cell \*matCellDef="let element"> {{element.name}} </mat-cell>
                           </ng-container>             
                          <!-- Position Column -->
                          <ng-container matColumnDef="position">
                            <mat-header-cell \*matHeaderCellDef> Position </mat-header-cell>
                            <mat-cell \*matCellDef="let element"> {{element.position}} </mat-cell>
                          </ng-container>              
                          <!-- Type Column -->
                          <ng-container matColumnDef="type">
                            <mat-header-cell \*matHeaderCellDef> Type </mat-header-cell>
                            <mat-cell \*matCellDef="let element"> {{element.type}} </mat-cell>
                          </ng-container>
                          <mat-header-row \*matHeaderRowDef="displayedColumns"></mat-header-row>
                          <mat-row \*matRowDef="let row; columns: displayedColumns;"></mat-row>
                        </mat-table>
                        </ign-data-table-container>
                      </div>
        </div></mat-tab>
    <mat-tab label="HTML"><div class="tab-height">
        <table style="width:100%">
            <p>&lt;div class="example-container mat-elevation-z8"&gt; </p>
            <p>&nbsp;&lt;ign-data-table-container [dataSource]="dataSource"&gt;</p>
            <p>&nbsp;&nbsp;&lt;mat-table #table [dataSource]="dataSource"&gt;</p>
            <p>&nbsp;&nbsp; &lt;!--- Note that these columns can be defined in any order.</p>
            <p>&nbsp;&nbsp; The actual rendered columns are set as a property on the row definition" --&gt;</p>
            <p>&nbsp;&nbsp; &lt;!-- Name Column --&gt;</p>
            <p>&nbsp;&nbsp; &lt;ng-container matColumnDef="name"&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&lt;mat-header-cell \*matHeaderCellDef&gt; Name &lt;/mat-header-cell&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&lt;mat-cell \*matCellDef="let element"&gt; element.name &lt;/mat-cell&gt;</p>
            <p>&nbsp;&nbsp; &lt;/ng-container&gt;</p>
            <p>&nbsp;&nbsp; &lt;!-- Position Column --&gt;</p>
            <p>&nbsp;&nbsp; &lt;ng-container matColumnDef="position"&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&lt;mat-header-cell \*matHeaderCellDef&gt; Position &lt;/mat-header-cell&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&lt;mat-cell \*matCellDef="let element"&gt; element.position &lt;/mat-cell&gt;</p>
            <p>&nbsp;&nbsp; &lt;/ng-container&gt;</p>
            <p>&nbsp;&nbsp; &lt;!-- Type Column --&gt;</p>
            <p>&nbsp;&nbsp; &lt;ng-container matColumnDef="type"&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&lt;mat-header-cell *matHeaderCellDef&gt; Type &lt;/mat-header-cell&gt;</p>
            <p>&nbsp;&nbsp;&nbsp;&nbsp;&lt;mat-cell \*matCellDef="let element"&gt; element.type &lt;/mat-cell&gt;</p>
            <p>&nbsp;&nbsp; &lt;mat-header-row \*matHeaderRowDef="displayedColumns"&gt;&lt;/mat-header-row&gt;</p>
            <p>&nbsp;&nbsp; &lt;mat-row \*matRowDef="let row; columns: displayedColumns;"&gt;&lt;/mat-row&gt;</p>
            <p>&nbsp;&nbsp;&lt;/mat-table&gt;</p>
            <p>&nbsp;&lt;/ign-data-table-container&gt;</p>
            <p>&lt;/div&gt;</p> 
        </table>
    </div></mat-tab>
</mat-tab-group>

## Content
* @Input() dataSource: MatTableDataSource 
    * Data being used for Material data table
* @Input() pageIndex: number                
    *  The index of the page 
* @Input() totalElements: number               
    * The total elements in a data table 
* @Input() pageSizeOptions: number[]            
    * Page size options to be made available within the paginator. 
    If none are provided, [5, 10, 15, 20, 50] will be used 
* @Input() pageSize: number                    
    * Default number of elements to display on a page. If none are specified, 10 will be displayed

## Style
* No styling is required for this component.

*** TypeScript *** 

 displayedColumns = ['name', 'position', 'type'];
 dataSource = new MatTableDataSource<Element>(ELEMENT_DATA);

*** ExternalTypeScript ***

export interface Element {
    name: string;
    position: string;
    type: string;
}

let ELEMENT_DATA: Element[] = [
    {name: 'Maham Boghani', position: 'Software Engineer', type: 'Human'},
    {name: 'Watson', position: 'Artificial Intelligence', type: 'Computer'},
    {name: 'Andy Avalos', position: 'Electrical Engineer', type: 'Human'},
    {name: 'Jensen', position: 'Director of Sleep', type: 'House Cat'},
    {name: 'Mariah Avalos', position: 'Software Engineer', type: 'Introvert'},
    {name: 'Emma Bear', position: 'Destroyer Of String', type: 'House Cat'},
    {name: 'Uchenna Ohaeto', position: 'Software Engineer', type: 'Human'},
    {name: 'Dennis', position: 'Being Repressed', type: 'Human'},
    {name: 'Sophia', position: 'Home Security', type: 'Dog'},
    {name: 'Brian Mahecha', position: 'Software Test Engineer', type: 'Human'},
    {name: 'Spock', position: 'First Officer, USS Enterprise', type: 'Vulcan'},
    {name: 'Daniel Weisberg', position: 'Software Engineer', type: 'Human'},
    {name: 'Inigo Montoya', position: 'Seeking Revenge', type: 'Human'},
    {name: 'Miguel Conde', position: 'Software Engineer', type: 'Human'},
    {name: 'Nikola Tesla', position: 'Inventor', type: 'Human'},
  
];
*** Imports ***

import { MatTableDataSource } from '@angular/material';