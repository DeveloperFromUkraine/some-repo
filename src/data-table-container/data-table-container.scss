@import '~ignite-common/styles/ignite';

$checkbox-length: 18px;
$hover-color: #EEEEEE;
$selection-color: #F5F5F5;

@mixin column-padding-first {
    padding-left: ign-padding(s);
}

@mixin column-padding-other {
    padding-left: 56px;
}

:host /deep/ .mat-header-cell  {
    &:nth-child(2) {
        @include column-padding-first();
    }
    &:nth-child(n+3)  {
        @include column-padding-other();
    }
}

:host /deep/ .mat-cell {
    &:nth-child(2) {
        @include column-padding-first();
    }
    &:nth-child(n+3) {
        @include column-padding-other();
    }
}

:host /deep/ .mat-column-select {
    // Needed for this: https://github.com/angular/material2/issues/8600
    overflow: initial;
}

:host /deep/ .mat-header-row {
    height: ign-spacing(xl);
}

:host /deep/ .mat-row {
    height: ign-spacing(l);

    &:hover {
        background: $hover-color;  
    }
    &::-moz-selection, &::-webkit-selection, &::selection {
        background: $selection-color;
    }
}

:host /deep/ .number-column {
    text-align: right;
}

:host /deep/ .mat-paginator {
    height: 56px;
}

:host /deep/ .mat-checkbox-inner-container {
    width: $checkbox-length;
    height: $checkbox-length;
}

:host /deep/ .mat-checkbox-checked.mat-accent .mat-checkbox-background, .mat-checkbox-indeterminate.mat-accent .mat-checkbox-background {
    background-color: ign-color(accent);
}