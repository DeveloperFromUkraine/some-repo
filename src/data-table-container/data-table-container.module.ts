import { NgModule } from '@angular/core';
import { MatPaginatorModule } from '@angular/material';
import { DataTableContainerComponent } from './data-table-container.component';

@NgModule({
  imports: [
    MatPaginatorModule,
  ],
  declarations: [
    DataTableContainerComponent,
  ],
  exports: [
    DataTableContainerComponent,
  ],
})
export class DataTableContainerModule {}
