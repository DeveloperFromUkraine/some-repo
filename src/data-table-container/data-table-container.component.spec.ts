import { ComponentFixture, TestBed } from '@angular/core/testing';
import { DataTableContainerComponent } from './data-table-container.component';
import { ComponentTest } from '../../test/test-bed/component';
import { DataTableContainerModule } from './data-table-container.module';
import { MatTableDataSource } from '@angular/material';
import { By } from '@angular/platform-browser';
import { DebugElement } from '@angular/core/src/debug/debug_node';

fdescribe('DataTableContainerComponent', () => {
    let fixture: ComponentFixture<DataTableContainerComponent>;
    let component: DataTableContainerComponent;
    let matTableDataSource: MatTableDataSource<any> = new MatTableDataSource();
    let de: DebugElement;
    let ne: HTMLElement;

    beforeEach(async () => {
        await ComponentTest.createTestBed([DataTableContainerModule]);
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(DataTableContainerComponent);
        component = fixture.componentInstance;
        component.dataSource = matTableDataSource;

        fixture.detectChanges();

        de = fixture.debugElement.query(By.css('mat-paginator'));
    });

    it('should set pageSize to given pageSize', () => {
        let pageSizeGiven = 25;
        component.pageSize = pageSizeGiven;
        ne = de.nativeElement;

        fixture.detectChanges();
        let pageSize = ne.getAttribute('ng-reflect-page-size');

        expect(pageSize).toBe(pageSizeGiven.toString());
    });

    it('should set pageSize to default pageSize', () => {
        let pageSizeDefault = 10;
        component.pageSize = null;
        ne = de.nativeElement;

        fixture.detectChanges();
        let pageSize = ne.getAttribute('ng-reflect-page-size');

        expect(pageSize).toBe(pageSizeDefault.toString());
    });

    it('should set pageSizeOptions to given options', () => {
        let pageSizeOptionsGiven = [1, 2, 3];
        component.pageSizeOptions = pageSizeOptionsGiven;
        ne = de.nativeElement;

        fixture.detectChanges();
        let pageSizeOptions = ne.getAttribute('ng-reflect-page-size-options');

        expect(pageSizeOptions).toBe(pageSizeOptionsGiven.toString());
    });

    it('should set pageSizeOptions to default page size options', () => {
        let pageSizeOptionsDefault = [5, 10, 15, 20, 50];
        component.pageSizeOptions = null;
        ne = de.nativeElement;

        fixture.detectChanges();
        let pageSizeOptions = ne.getAttribute('ng-reflect-page-size-options');

        expect(pageSizeOptions).toBe(pageSizeOptionsDefault.toString());
    });

    it('should call emitPageEvents if page event is triggered', () => {
        jest.spyOn(component, 'emitPageEvents');

        de.triggerEventHandler('page', null);

        expect(component.emitPageEvents).toHaveBeenCalledTimes(1);
    });

    it('should emit paginatorEvent if page event is triggered', () => {
        let spy = jest.fn();

        component.paginatorEvents$.subscribe(spy);
        de.triggerEventHandler('page', null);

        expect(spy).toHaveBeenCalled();
    });

    it('should match snapshot', () => {
        expect(fixture).toMatchSnapshot();
    });
});
