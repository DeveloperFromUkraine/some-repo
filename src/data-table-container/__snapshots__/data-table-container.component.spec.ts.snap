// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Container should match snapshot 1`] = `
<ign-data-table-container
  dataSource={[Function MatTableDataSource]}
  paginator={[Function MatPaginator]}
  paginatorEvents$={[Function BehaviorSubject]}
>
  
  
  <mat-paginator
    class="mat-paginator"
    ng-reflect-page-size="10"
    ng-reflect-page-size-options="5,10,15,20,50"
  >
    <div
      class="mat-paginator-container"
    >
      
      <div
        class="mat-paginator-page-size ng-star-inserted"
      >
        <div
          class="mat-paginator-page-size-label"
        >
          Items per page:
        </div>
        
        <mat-form-field
          class="mat-paginator-page-size-select mat-input-container mat-form-field ng-tns-c37-0 mat-primary mat-form-field-type-mat-select mat-form-field-can-float mat-form-field-hide-placeholder ng-star-inserted"
        >
          <div
            class="mat-input-wrapper mat-form-field-wrapper"
          >
            <div
              class="mat-input-flex mat-form-field-flex"
            >
              
              <div
                class="mat-input-infix mat-form-field-infix"
              >
                <mat-select
<<<<<<< HEAD
                  tabindex="0"
                  class="mat-select ng-tns-c38-1 ng-star-inserted"
                  ng-reflect-value="10"
                  ng-reflect-aria-label="Items per page:"
                  id="mat-select-6"
                  role="listbox"
                  aria-label="Items per page:"
                  aria-required="false"
                  aria-disabled="false"
                  aria-invalid="false"
=======
                  aria-disabled="false"
                  aria-invalid="false"
                  aria-label="Items per page:"
>>>>>>> develop
                  aria-multiselectable="false"
                  aria-required="false"
                  class="mat-select ng-tns-c38-1 ng-star-inserted"
                  id="mat-select-6"
                  ng-reflect-aria-label="Items per page:"
                  ng-reflect-value="10"
                  role="listbox"
                  tabindex="0"
                >
                  <div
                    aria-hidden="true"
                    cdk-overlay-origin=""
                    class="mat-select-trigger"
                  >
                    <div
                      class="mat-select-value"
                      ng-reflect-ng-switch="true"
                    >
<<<<<<< HEAD
                      <span
                        class="mat-select-placeholder ng-tns-c38-1 ng-star-inserted"
                      >
                        
                      </span>
=======
                      
                      <span
                        class="mat-select-placeholder ng-tns-c38-1 ng-star-inserted"
                      >
                        Â 
                      </span>
                      
>>>>>>> develop
                    </div>
                    <div
                      class="mat-select-arrow-wrapper"
                    >
                      <div
                        class="mat-select-arrow"
                      />
                    </div>
                  </div>
                  
                </mat-select>
                <span
                  class="mat-form-field-label-wrapper mat-input-placeholder-wrapper mat-form-field-placeholder-wrapper"
                >
                  
                </span>
              </div>
              
            </div>
            <div
              class="mat-input-underline mat-form-field-underline"
            >
              <span
                class="mat-input-ripple mat-form-field-ripple"
              />
            </div>
            <div
              class="mat-input-subscript-wrapper mat-form-field-subscript-wrapper"
              ng-reflect-ng-switch="hint"
            >
              
              
              <div
                class="mat-input-hint-wrapper mat-form-field-hint-wrapper ng-tns-c37-0 ng-trigger ng-trigger-transitionMessages ng-star-inserted"
                style="opacity: 1;"
              >
                
                <div
                  class="mat-input-hint-spacer mat-form-field-hint-spacer"
                />
              </div>
            </div>
          </div>
        </mat-form-field>
        
      </div>
      <div
        class="mat-paginator-range-actions"
      >
        <div
          class="mat-paginator-range-label"
        >
          0 of 0
        </div>
        
        <button
          aria-describedby="cdk-describedby-message-12"
          aria-label="Previous page"
          cdk-describedby-host=""
          class="mat-paginator-navigation-previous mat-icon-button"
          disabled=""
          mat-icon-button=""
          ng-reflect-disabled="true"
          ng-reflect-message="Previous page"
          ng-reflect-position="above"
          type="button"
        >
          <span
            class="mat-button-wrapper"
          >
            <div
              class="mat-paginator-decrement"
            />
          </span>
          <div
            class="mat-button-ripple mat-ripple mat-button-ripple-round"
            matripple=""
            ng-reflect-centered="true"
            ng-reflect-disabled="true"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        <button
          aria-describedby="cdk-describedby-message-13"
          aria-label="Next page"
          cdk-describedby-host=""
          class="mat-paginator-navigation-next mat-icon-button"
          disabled=""
          mat-icon-button=""
          ng-reflect-disabled="true"
          ng-reflect-message="Next page"
          ng-reflect-position="above"
          type="button"
        >
          <span
            class="mat-button-wrapper"
          >
            <div
              class="mat-paginator-increment"
            />
          </span>
          <div
            class="mat-button-ripple mat-ripple mat-button-ripple-round"
            matripple=""
            ng-reflect-centered="true"
            ng-reflect-disabled="true"
            ng-reflect-trigger="[object HTMLButtonElement]"
          />
          <div
            class="mat-button-focus-overlay"
          />
        </button>
        
      </div>
    </div>
  </mat-paginator>
  
</ign-data-table-container>
`;
